src/shipit_code_coverage/shipit_code_coverage/artifacts.py

```diff
@@ -124,6 +79,29 @@ def download_all(self):
             if taskcluster.is_coverage_task(task)
         ]
 
+        # Choose best tasks to download (e.g. 'completed' is better than 'failed')
+        download_tasks = {}
+        for test_task in test_tasks:
+            status = test_task['status']['state']
+            assert status in ALL_STATUSES
+
+            chunk_name = taskcluster.get_chunk(test_task['task']['metadata']['name'])
+            platform_name = taskcluster.get_platform(test_task['task']['metadata']['name'])
+            # Ignore awsy and talos as they aren't actually suites of tests.
+            if any(to_ignore in chunk_name for to_ignore in self.suites_to_ignore):
+                continue
+
+            if (chunk_name, platform_name) not in download_tasks:
+                # If the chunk hasn't been downloaded before, this is obviously the best task
+                # to download it from.
+                download_tasks[(chunk_name, platform_name)] = test_task
+            else:
+                # Otherwise, compare the status of this task with the previously selected task.
+                prev_task = download_tasks[(chunk_name, platform_name)]
```

Comment:
Nit: I would invert both cases of the parent  `if` so that the more complex case is above. But that's my own internal brain logic here